Return-Path: <linux-wireless-owner@vger.kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on dual
X-Spam-Level: 
X-Spam-Status: No, score=0.9 required=5.0 tests=BAYES_00,FREEMAIL_FROM,
	KB_DATE_CONTAINS_TAB autolearn=no version=3.3.1
X-Original-To: maks@dual
Delivered-To: maks@dual
Received: from dual (localhost.localdomain [127.0.0.1])
	by dual (Postfix) with ESMTP id ED48524045
	for <maks@dual>; Fri, 23 Apr 2010 04:14:59 +0200 (CEST)
X-Original-To: max@stro.at
Delivered-To: max@stro.at
Received: from baikonur.stro.at [213.239.196.228]
	by dual with POP3 (fetchmail-6.3.16)
	for <maks@dual> (single-drop); Fri, 23 Apr 2010 04:14:59 +0200 (CEST)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by baikonur.stro.at (Postfix) with ESMTP id 407AB5C001
	for <max@stro.at>; Thu, 22 Apr 2010 19:44:39 +0200 (CEST)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
	id S1755433Ab0DVRwc (ORCPT <rfc822;max@stro.at>);
	Thu, 22 Apr 2010 13:52:32 -0400
Received: from mail-vw0-f46.google.com ([209.85.212.46]:52767 "EHLO
	mail-vw0-f46.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
	with ESMTP id S1755072Ab0DVRwb (ORCPT
	<rfc822;linux-wireless@vger.kernel.org>);
	Thu, 22 Apr 2010 13:52:31 -0400
Received: by vws14 with SMTP id 14so38715vws.19
        for <linux-wireless@vger.kernel.org>; Thu, 22 Apr 2010 10:52:29 -0700 (PDT)
DKIM-Signature:	v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=googlemail.com; s=gamma;
        h=domainkey-signature:received:received:received:received:received
         :from:to:subject:date:user-agent:cc:mime-version:x-length:x-uid
         :content-type:content-transfer-encoding:message-id;
        bh=rH+Y7riac85RIiXG6qbakuXBSY790hWoyH0ONriCSeg=;
        b=Ykzs36TepjY445cGIr5IH3C+knJyI56PTTlTKWlRkO098l3gXlGleAI5fn2s5sivNa
         F5+SpVSoGspL0FT7yHTYb+CnZ+6ZyzshVgudsBPAakjfrZNlBYpEub3PAgddsugf9A75
         j1t9aq3UhlHL13NLQQg1mwfWsN5hUSbfjKwZQ=
DomainKey-Signature: a=rsa-sha1; c=nofws;
        d=googlemail.com; s=gamma;
        h=from:to:subject:date:user-agent:cc:mime-version:x-length:x-uid
         :content-type:content-transfer-encoding:message-id;
        b=ZixBFP6qSmufbPzN9DU5JjG83DMDIZFFEClBzUKD1bszfbpbEy3fsXLEFwgDchVGf2
         VpqO7MVL6R70pPqT4CGb/k2XnaFpVM3lQZK695ehjHK+X/g0nPaprPYrZu/TbPii3Ieg
         3Dj6OuAw0/tMNg5FUfKbIlLct8v3HfpJmNVwE=
Received: by 10.220.126.153 with SMTP id c25mr1130560vcs.140.1271958749500;
        Thu, 22 Apr 2010 10:52:29 -0700 (PDT)
Received: from blech.mobile ([72.14.241.41])
        by mx.google.com with ESMTPS id i29sm814619vcr.12.2010.04.22.10.52.27
        (version=TLSv1/SSLv3 cipher=RC4-MD5);
        Thu, 22 Apr 2010 10:52:28 -0700 (PDT)
Received: from blech.mobile ([127.0.0.1])
	by localhost (localhost [127.0.0.1]) (amavisd-new, port 10024)
	with ESMTP id bUG7mXcCH9aE; Thu, 22 Apr 2010 19:52:17 +0200 (CEST)
Received: from blech.mobile (localhost [127.0.0.1])
	by blech.mobile (Postfix) with ESMTP id ADBA2342417;
	Thu, 22 Apr 2010 19:52:17 +0200 (CEST)
From: Christian Lamparter <chunkeey@googlemail.com>
To: linux-wireless@vger.kernel.org
Subject: [PATCH 1/2] p54pci: fix bugs in p54p_check_tx_ring
Date:	Thu, 22 Apr 2010 19:52:16 +0200
User-Agent: KMail/1.12.4 (Linux/2.6.34-rc5-uber-wl; KDE/4.3.4; x86_64; ; )
Cc: linville@tuxdriver.com, hdegoede@redhat.com
MIME-Version: 1.0
X-Length: 2052
X-UID:	75
Content-Type: Text/Plain;
  charset="iso-8859-1"
Content-Transfer-Encoding: 7bit
Message-Id: <201004221952.16857.chunkeey@googlemail.com>
Sender: linux-wireless-owner@vger.kernel.org
Precedence: bulk
List-ID: <linux-wireless.vger.kernel.org>
X-Mailing-List:	linux-wireless@vger.kernel.org
Content-Length: 1159

From: Hans de Goede <hdegoede@redhat.com>

Hans de Goede identified a bug in p54p_check_tx_ring:

there are two ring indices. 1 => tx data and 3 => tx management.
But the old code had a constant "1" and this resulted in spurious
dma unmapping failures. 

Cc: stable@kernel.org
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=583623
Bug-Identified-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
diff --git a/drivers/net/wireless/p54/p54pci.c b/drivers/net/wireless/p54/p54pci.c
index 679da7e..cffe2f2 100644
--- a/drivers/net/wireless/p54/p54pci.c
+++ b/drivers/net/wireless/p54/p54pci.c
@@ -246,7 +246,7 @@ static void p54p_check_tx_ring(struct ieee80211_hw *dev, u32 *index,
 	u32 idx, i;
 
 	i = (*index) % ring_limit;
-	(*index) = idx = le32_to_cpu(ring_control->device_idx[1]);
+	(*index) = idx = le32_to_cpu(ring_control->device_idx[ring_index]);
 	idx %= ring_limit;
 
 	while (i != idx) {
--
To unsubscribe from this list: send the line "unsubscribe linux-wireless" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

